packages/angular_devkit/schematics/tools/node-module-engine-host.ts

```diff
@@ -102,7 +102,11 @@ export class NodeModulesEngineHost extends FileSystemEngineHostBase {
         return collectionPath;
       }
     } catch (e) {
+      if (!(e instanceof FileDoesNotExistException)) {
+        throw new InvalidCollectionJsonException(name, collectionPath);
```

Comment:
Could you just rethrow the error instead? The sub error can be a JSON error which contains more information (line and character), or another error.

Reply:
I initially was gonna do did that, the only drawback was the in the error there won’t be information about which files cause the JSON error. 

An alternative would be that `InvalidCollectionJsonException` would accept another parameter, and eventually if error.message is present it will be concatinated with original message that `InvalidCollectionJsonException` is showing to the user.

What do yo think?

Reply:
Something similar to this error https://github.com/angular/angular-cli/issues/11861 with no context of which file is the problem.

Reply:
I think either solution would be better:

1. An alternative would be that InvalidCollectionJsonException would accept another parameter of type `InvalidJsonException` and transfer over the info, or
2. A separate new `InvalidJsonFileException` which takes a path and the json exception.

I'm fine with either. I have a slight preference to 1.

Reply:
I also prefer for the first option. I have pushed an updated. When you said `transfer over the info` I was not sure if you were referring the error message or stack or both. For the time being I did the message.